kolibri/core/errorreports/models.py

```diff
@@ -84,34 +79,35 @@ def save(self, *args, **kwargs):
         super().save(*args, **kwargs)
 
     @classmethod
-    def insert_or_update_error(
-        cls, category, error_message, traceback, context, request_time_to_error=None
-    ):
-        if not getattr(settings, "DEVELOPER_MODE", None):
-            error, created = cls.objects.get_or_create(
+    def insert_or_update_error(cls, category, error_message, traceback, context):
+        if getattr(settings, "DEVELOPER_MODE", False):
+            logger.info(
+                "ErrorReports: Database not updated, as DEVELOPER_MODE is True."
+            )
+            return
+        error_report = cls.objects.filter(
```

Comment:
Could still use `get_or_create` here - you can pass information that you don't want to filter by, but do want added to a newly created object by using the `defaults` dict: https://docs.djangoproject.com/en/3.2/ref/models/querysets/#get-or-create

The defaults dict will be used to add additional data to the object if it is newly created, but otherwise will be discarded.

Reply:
Thank you, it eliminates extra else statement!